08.ls_object/file_detail.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class FileDetail
+  PERMISSION_PATTERNS = {
+    '0' => '---',
+    '1' => '--x',
+    '2' => '-w-',
+    '3' => '-wx',
+    '4' => 'r--',
+    '5' => 'r-x',
+    '6' => 'rw-',
+    '7' => 'rwx'
+  }.freeze
+
+  def initialize(file_name)
+    @file_name = file_name
+    @stat = File.stat(file_name)
+  end
+
+  def type_and_permission
+    case @stat.ftype
+    when 'file'
+      file_type = '-'
+    when 'directory'
+      file_type = 'd'
+    when 'link'
+      file_type = 'l'
+    end
+    permission_number = @stat.mode.to_s(8).split('').last(3).join('')
+    permission_symbol = permission_number.gsub(/[01234567]/, PERMISSION_PATTERNS)
```

Comment:
正規表現の文字クラスの部分は `[0-7]` で良さそうです。

```suggestion
    permission_symbol = permission_number.gsub(/[0-7]/, PERMISSION_PATTERNS)
```

Reply:
修正しました！